Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleTableView.h

```diff
@@ -36,11 +47,33 @@ namespace ShaderManagementConsole
         void OnCellSelected(int row, int column, int previousRow, int previousColumn);
         void OnCellChanged(int row, int column);
 
+        void ShowContextMenu(const QPoint& pos);
+
+        void keyPressEvent(QKeyEvent*) override;
+        void mousePressEvent(QMouseEvent* event) override;
+
+        enum RebuildMode { KeepAsIs, CallOnModified };
+        void TransferViewModelToModel(RebuildMode);
+
         const AZ::Crc32 m_toolId = {};
         const AZ::Uuid m_documentId = AZ::Uuid::CreateNull();
         AZ::RPI::ShaderVariantListSourceData m_shaderVariantListSourceData;
         AZStd::vector<AZ::RPI::ShaderOptionDescriptor> m_shaderOptionDescriptors;
         size_t m_shaderVariantCount = {};
         size_t m_shaderOptionCount = {};
+        ColumnSortMode m_columnSortMode = Cost;
+    };
+
+    class ShaderManagementConsoleContainer : public QVBoxLayout
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(ShaderManagementConsoleContainer, AZ::SystemAllocator);
+
+        ShaderManagementConsoleContainer(QWidget* container, const AZ::Crc32& toolId, const AZ::Uuid& documentId, QWidget* parent);
+
+        ShaderManagementConsoleTableView m_tableView;
```

Comment:
about this, all other code places have dynamic per object allocations, and raw pointer passing to Qt. I find this disturbing but if that's Qt's culture, that class will need to be changed from storing its own widgets to simply newing them in the cpp